src/workerd/api/web-socket.c++

```diff
@@ -328,12 +328,13 @@ kj::Promise<DeferredProxy<void>> WebSocket::couple(kj::Own<kj::WebSocket> other)
     // Since the WebSocket is terminated locally, we need the IoContext to stay live while
     // it is active.
     co_await promise;
-    co_return DeferredProxy<void> { kj::READY_NOW };
+    co_return;
   } else {
     // Since the WebSocket is just proxying through, we can do the pump in a deferred proxy task.
     // Note that we don't need to (and can't) register any pending events in this case since the
     // IoContext is free to go away at this point.
-    co_return DeferredProxy<void> { kj::mv(promise) };
+    co_yield BEGIN_DEFERRED_PROXYING;
```

Comment:
Definitely like this syntax but it would likely be good to add some comments around it to help new folks understand it.

Reply:
I added a comment.